src/components/NavMenu.tsx

```diff
@@ -0,0 +1,312 @@
+import React, { useRef, useState, useEffect, useCallback } from 'react';
+import {
+	SafeAreaView,
+	View,
+	FlatList,
+	Text,
+	Image,
+	TouchableOpacity,
+	Animated,
+	PanResponder,
+	Easing,
+	StyleSheet,
+	Dimensions,
+	ImageSourcePropType,
+} from 'react-native';
+import { Icon } from 'react-native-elements';
+
+import Toggle from './Toggle';
+
+interface HeaderProps {
+	imgSrc: ImageSourcePropType;
+	title: string;
+	subtitles: string[];
+}
+
+const MenuHeader = ({ imgSrc, title, subtitles }: HeaderProps) => {
+	return (
+		<View style={styles.menuHeader}>
+			<Image source={imgSrc} style={imageStyles.headerImage} />
+			<View>
+				<Text style={styles.text}>{title}</Text>
+				{subtitles.map((sub) => (
+					<Text style={{ fontSize: 12 }} key={sub}>
+						{sub}
+					</Text>
+				))}
+			</View>
+		</View>
+	);
+};
+
+interface Link {
+	text: string;
+	route: string;
+	iconName: string;
+}
+
+interface MenuProps {
+	header: HeaderProps;
+	links: Link[];
+	widthPercent: number;
+	toggleState: boolean;
+	onToggleChange(val: boolean): void;
+	openState: boolean;
+	onOpenChange(val: boolean): void;
+}
+
+const Menu = ({
+	header,
+	links,
+	widthPercent,
+	toggleState,
+	onToggleChange,
+	openState,
+	onOpenChange,
+}: MenuProps) => {
+	// All translations are offset by `width`,
+	// to account for the visual buffer on the left of the menu if the user swipes too far.
+	// (the full width of the sidebar menu is `2 * width`)
```

Comment:
The sidebar does stop if you try and swipe past the whole buffer width (in the `panResponder`), but the buffer just makes swiping feel more natural since there's some room for the menu to move past its edge. 

But I agree, having the entire width as a buffer is kinda overkill, I'll just make it a constant buffer of like 50px.